azerothcore/azerothcore-wotlk
# feat: add argument to .character rename to add name to reserved_name

## Changes Proposed:
- Add an optional argument `[reserveName]` to the `.character rename` command. If this argument is `1`, then the current player's name is added to the list of reserved names, so that it becomes unavailable.

## Tests Performed:

- Applied SQL and C++ fixes, tested on Windows 10 server and clients

## How to Test the Changes:


1. Select a character in-game and do `.char rename 1` or `.char rename Yournamehere 1`
2. Try to create a character with that name, verify that it says "This name is unavailable"
3. Play around with the command, make sure everything else still works as intended

## Known Issues and TODO List:


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
